src/EditModel/objectActions.js

```diff
@@ -84,8 +84,45 @@ const afterSaveHacks = {
 
         return Observable.fromPromise(Promise.all([removePromises, savePromises]));
     },
+    attribute: function attributeAfterSave() {
+        // When an attribute is added/edited/deleted, the attributes of all d2 modelDefinitions
+        // need to be updated manually, because d2 treats these as static properties
+        const d2Promise = getInstance();
+        const attributePromise = d2Promise.then(d2 => {
+            const api = d2.Api.getApi();
+            const queryParams = { fields: ':all,optionSet[:all,options[:all]]', paging: false };
+            return api.get('attributes', queryParams);
+        })
+
+        return Observable.fromPromise(Promise.all([d2Promise, attributePromise])
```

Comment:
I think this would've been cleaner if you made the function async. Then you would not need promise.all, as you would have d2 in the scope. with ```const d2 = await getInstance()``` .

I guess that's a different style than the other afterSaveHacks, though.